src/components/marquee/index.tsx

```diff
@@ -0,0 +1,118 @@
+
+const DEFAULT_DURATION = 3000;
+const DEFAULT_DURATION_PER_WORD = 250;
+
+function Marquee(props: MarqueeProps) {
+  const {label, labelStyle, direction = MarqueeDirections.LEFT, duration, numberOfReps = -1, containerStyle} = props;
+
+  const calcDuration = () => {
+    const numOfWords = label.split(' ').length;
+    return DEFAULT_DURATION + DEFAULT_DURATION_PER_WORD * numOfWords;
+  };
+
+  const isHorizontal = direction === MarqueeDirections.LEFT || direction === MarqueeDirections.RIGHT;
+  const fixedDuration = isHorizontal ? duration || calcDuration() : duration || DEFAULT_DURATION;
+
+  const [viewLayout, setViewLayout] = useState<LayoutRectangle | undefined>(undefined);
+  const [textLayout, setTextLayout] = useState<LayoutRectangle | undefined>(undefined);
+
+  const offset = useSharedValue<number | undefined>(undefined);
+
+  let initialOffset = 0;
+  let axisX = false;
+  let axisY = false;
+
+  if (direction === MarqueeDirections.RIGHT || direction === MarqueeDirections.LEFT) {
+    axisX = true;
+  } else {
+    axisY = true;
+  }
+
+  const onLayoutView = (event: LayoutChangeEvent) => {
+    setViewLayout(event.nativeEvent.layout);
+  };
+
+  const onLayoutText = (event: LayoutChangeEvent) => {
+    setTextLayout(event.nativeEvent.layout);
+  };
+
+  const startAnimation = (fromValue: number, toValue: number, backToValue: number) => {
+    initialOffset = fromValue;
+    offset.value = initialOffset;
+
+    offset.value = withRepeat(withTiming(toValue, {duration: fixedDuration, easing: Easing.linear}),
+      numberOfReps,
+      false,
+      finished => {
+        if (finished) {
+          offset.value = initialOffset;
+          offset.value = withTiming(backToValue, {
+            duration: 2000
+          });
+        }
+      });
+  };
+
+  const translateStyle = useAnimatedStyle(() => {
+    if (offset.value) {
+      return {
+        transform: [{translateX: axisX ? offset.value : 0}, {translateY: axisY ? offset.value : 0}]
+      };
+    }
+    return {};
+  });
+
+  useEffect(() => {
+    if (viewLayout && textLayout) {
+      switch (direction) {
+        case MarqueeDirections.RIGHT:
+          startAnimation(-textLayout.width, viewLayout.width, 0);
+          break;
+        case MarqueeDirections.LEFT:
+          startAnimation(viewLayout?.width, -textLayout.width, viewLayout.width - textLayout.width);
+          break;
+        case MarqueeDirections.UP:
+          startAnimation(viewLayout.height, -textLayout.height, viewLayout.height - textLayout.height);
+          break;
+        case MarqueeDirections.DOWN:
+          startAnimation(-textLayout.height, viewLayout.height, 0);
+          break;
+      }
+    }
+  }, [viewLayout, textLayout]);
+
+  return (
+    <View style={[styles.container, containerStyle]} onLayout={onLayoutView}>
+      <View
+        reanimated
+        style={[
+          {position: 'absolute', width: !isHorizontal || textLayout?.width ? textLayout?.width : '400%'},
```

Comment:
consider moving this part to useAnimatedStyle. 
It might not be animated, but it will unified the style into a single object.